pkg/apis/core/v1beta1/kubefedconfig_types.go

```diff
@@ -80,19 +91,24 @@ const (
 
 type ClusterHealthCheckConfig struct {
 	// How often to monitor the cluster health (in seconds).
-	PeriodSeconds int64 `json:"periodSeconds"`
```

Comment:
Not the fault of this PR, but why should time in the health check config be specified as an integer representing seconds but in the leader election config as a duration?  This seems inconsistent to me.

cc: @shashidharatd

Reply:
yes, there is inconsistency. it was thought like if we explicitly put unit in the option/flag name itself there will be no confusion while providing the config. Otherwise user has to explicitly provide something like "10s" to specify 10 seconds. If user misses the "s" it is still a valid configuration (in nanoseconds) which leads to undesired value getting configured.

Reply:
Fair enough. Why, though, is this important for cluster health check but not leader election?  And which is more consistent with user expectations?

Reply:
Mostly, it is like that because leader election module uses the `Duration` type and if we change them to `int` we have to convert and pass the parameters https://github.com/kubernetes-sigs/kubefed/blob/master/cmd/controller-manager/app/leaderelection/leaderelection.go#L72

Leader election module is used by multiple components and i think whoever configures them might already be aware of such an issue :) Now we need to make a choice ;)

Reply:
I think if leader election uses duration across components, is there a reason we shouldn't be consistent and also use duration for leader election?  And if we use duration for that, why wouldn't we use duration for the health check config?

Reply:
I am fine with that choice.

Reply:
If we move to use `Duration` I think the confusing thing for users is the use of the unit in the name. So if we switch, I think we should remove the unit in the name for this and for `TimeoutSeconds` with a comment expressing it should be in seconds. Then we'll need to perform validation to make sure it is in seconds. WDYT?

Reply:
I'm +1 on using duration and removing the unit from the name.  I'm less clear on why we'd need to validate that the value was in seconds, is that a CRD thing?  Whatever duration is provided could be converted to seconds.

Reply:
It's not a CRD thing. It's more of a granularity thing. I figured the reason for requiring seconds is to allow a sane amount of interval delay granularity between cluster health checks as anything less than 1 second may not be reasonable.

Reply:
I think it's entirely dependent on context.  Maybe partial seconds (e.g. 1.5) would be useful in some circumstances?  I think setting sane defaults and documenting how and why to change those defaults is a reasonable alternative to strict validation.

Reply:
I have no issues with being less opinionated about the level of granularity we allow. We can always enforce a certain level in validation. I think this change should be done in a follow-up PR though since this PR focuses on optional fields.

Reply:
Please file an issue in the milestone that tracks the transition to use duration for all time fields.